File: src/main/java/tech/zolhungaj/amqapi/servercommands/globalstate/FriendOnlineChange.java
package tech.zolhungaj.amqapi.servercommands.globalstate;

import com.squareup.moshi.Json;
import tech.zolhungaj.amqapi.servercommands.Command;
import tech.zolhungaj.amqapi.servercommands.CommandTypeOld;

public record FriendOnlineChange (
        @Json(name = "name") String playerName,
        Boolean online
) implements Command {
    @Override
    public String commandName() {
        return CommandTypeOld.FRIEND_STATE_UPDATE.commandName;
    }
}

File: src/main/java/tech/zolhungaj/amqapi/servercommands/globalstate/FriendOnlineChange.kt
package tech.zolhungaj.amqapi.servercommands.globalstate;

import com.squareup.moshi.Json;
import tech.zolhungaj.amqapi.servercommands.CommandType;

@CommandType("friend state change")
public record FriendOnlineChange (
        @Json(name = "name") String playerName,
        Boolean online
){}
